teamcity/trigger_schedule_options.go

```diff
@@ -2,16 +2,16 @@ package teamcity
 
 //TriggerScheduleOptions represent options for configuring a scheduled build trigger
 type TriggerScheduleOptions struct {
-	QueueOptimization                   bool                       `prop:"enableQueueOptimization" force:""`
-	PromoteWatchedBuild                 bool                       `prop:"promoteWatchedBuild" force:""`
-	BuildWithPendingChangesOnly         bool                       `prop:"triggerBuildWithPendingChangesOnly" force:""`
-	TriggerIfWatchedBuildChanges        bool                       `prop:"triggerBuildIfWatchedBuildChanges"`
-	BuildOnAllCompatibleAgents          bool                       `prop:"triggerBuildOnAllCompatibleAgents"`
-	RevisionRuleSourceBuildID           string                     `prop:"revisionRuleDependsOn"`
-	RevisionRule                        ArtifactDependencyRevision `prop:"revisionRule"`
-	RevisionRuleBuildBranch             string                     `prop:"revisionRuleBuildBranch"`
-	EnforceCleanCheckout                bool                       `prop:"enforceCleanCheckout"`
-	EnforceCleanCheckoutForDependencies bool                       `prop:"enforceCleanCheckoutForDependencies"`
+	QueueOptimization            bool                       `prop:"enableQueueOptimization" force:""`
+	PromoteWatchedBuild          bool                       `prop:"promoteWatchedBuild" force:""`
+	BuildWithPendingChangesOnly  bool                       `prop:"triggerBuildWithPendingChangesOnly" force:""`
+	TriggerIfWatchedBuildChanges bool                       `prop:"triggerBuildIfWatchedBuildChanges"`
+	BuildOnAllCompatibleAgents   bool                       `prop:"triggerBuildOnAllCompatibleAgents"`
+	RevisionRuleSourceBuildID    string                     `prop:"revisionRuleDependsOn"`
+	RevisionRule                 ArtifactDependencyRevision `prop:"revisionRule"`
+	RevisionRuleBuildBranch      string                     `prop:"revisionRuleBuildBranch"`
+
+	BuildCustomization *BuildTriggerCustomization
```

Comment:
Not actually, it's used for serializing properties, and buildCustomization is not one of them. See https://github.com/yext/go-teamcity/blob/yext/teamcity/properties.go#L193
And an example trigger structure: https://teamcity-server.yext.use4a.cnc-sbx-o2cwhite.net/app/rest/buildTypes/Thundersnow_KhanQuotes_Commit/triggers

Reply:
Currently the TriggerSchedule struct is a logic representation of the actual teamcity resource, it was not clear to me the line between a TriggerScheduleOption and a field belong to TriggerSchedule, because some fields on Trigger Schedule like cron numbers and some options like triggerBuildIfWatchedBuildChanges are both listed as properties.

From the current implementation I was thinking the fields in TriggerSchedule are mostly "defining" the schedule, like the cron rules, while the fields in options are customizable options like `triggerBuildIfWatchedBuildChanges`, and buildCustomization might logically be closer there. They can arguably be flatten to only one struct if we want to invest the time to refactor. 

Any other thoughts on buildCustomization might be better to be in TriggerSchedule?

Reply:
I don't feel strongly about it, so it sounds good to me as it is today. As the terraform state is independent of this refactoring, we can revisit if need be.